kernel/src/common/buffer.rs

```diff
@@ -0,0 +1,151 @@
+//! Defines a Buffer type which can be used to pass a section of a larger
+//! buffer but still get the entire buffer back in a callback
+//!
+//! Author: Amit Levy
+
+use core::ops::{Bound, Range, RangeBounds};
+use core::ops::{Index, IndexMut};
+use core::slice::SliceIndex;
+
+pub struct Buffer<'a, T> {
```

Comment:
Maybe `SliceableBuffer` as a name? Or `LeaseableBuffer`? In any case I think we need something that signals this is not an ordinary buffer.

Reply:
`LeasableBuffer` sounds good to me, what should I do with the commented out tests in this file? I could just delete them, but they are sort of useful to have around in case anyone ever wanted to modify this interface and check that it still works.